lib/httplib/httplib.go

```diff
@@ -249,14 +251,25 @@ func RewritePaths(next http.Handler, rewrites ...RewritePair) http.Handler {
 	})
 }
 
-// SafeRedirect performs a relative redirect to the URI part of the provided redirect URL
-func SafeRedirect(w http.ResponseWriter, r *http.Request, redirectURL string) error {
+// OriginLocalRedirectURI will take an incoming URL including optionally the host and schema and provide the URI
+// associated with the URL.  Additionally, it will ensure that the URI does not include any techniques potentially
```

Comment:
On first read I was very confused by what "the URI associated with the URL" meant.

Would it make more sense to call this function something like `ExtractPath`?

Reply:
Unclear to me if `ExtractPath` is a better name.  This specifically is trying to extract the path in a way that is safe for redirects when the origin must remain consistent.  So I tried to have the word `Redirect` in it (since that's the only known use case), and I tried to make sure that the `Origin` concept was also in the name.

`ExtractPath` sounds too generic IMO, and does not indicate the additional checks put here.

`ExtractOriginLocalSafePath` I think could be a good option, but I don't feel strongly about it.  Do you like that naming at all?

Reply:
Hmm, yeah I see that it's doing quite a bit more than that.

I wonder if `httplib` is the right place for this - it seems to be a pretty specialized function written specifically to handle SSO redirects. Should it be located closer to the code that handles SSO?

Reply:
No strong preference on my end.  I put it in `httplib` because theoretically this is useful for any origin local redirects (though SSO was the only place I could find where we currently need this).  I suspect this fairly generic quality is why the prior (unused) `SafeRedirect` also lived in `httplib`